proversity_reports_script/request_report.py

```diff
@@ -44,32 +44,92 @@ def init_report_pipeline(self, *args, **kwargs):
+                request_headers=self.get_request_headers(),
+            )
+        )
+
+    def get_report_generation_data(self):
+        """
+        Return the response data from the report generation request.
+
+        Since API v0 does not support paging, it returns the plain response data.
+
+        For API v1, to avoid requesting data for a large set of courses (and probably get a 504 error),
+        the report generation request is divided into groups of courses
+        determined by EXTRA_DATA['MAX_COURSES_PER_REQUEST'] from the report configuration.
+        Each response data is collected into one object.
+
+        Returns:
+            v0:
+                {
+                    "message": "The task with id = {task id} has been initialize.",
+                    "state_url": "http://lms-domain/proversity-reports/api/v0/get-report-data?task_id={task id}",
+                    "success": true
+                }
+            v1:
+                {
+                    "data": [
+                        "course-id-0": [
+                            "http://lms-domain/proversity-reports/api/v1/get-report-data?task_id={task id}"
+                        ],
+                        "course-id-1": [
+                            "http://lms-domain/proversity-reports/api/v1/get-report-data?task_id={task id}"
+                            "http://lms-domain/proversity-reports/api/v1/get-report-data?task_id={task id}"
+                        ]...
+                    ]
+                }
+        """
         initial_report_request_url, report_generation_request_data = self.report_generation_request_data()
         extra_request_data = self.report_settings.get('EXTRA_REQUEST_DATA', {})
-        request_headers = {
-            'Authorization': 'Bearer {}'.format(self.settings.get('OPEN_EDX_OAUTH_TOKEN', '')),
-            'Content-Type': 'application/json',
-        }
 
         if not initial_report_request_url:
             print('Report URL was not provided.')
             exit()
 
-        print('Report generation requested to: {}'.format(initial_report_request_url))
-        initial_report_generation_data = request_handler(
-            request_url=initial_report_request_url,
-            request_data=report_generation_request_data,
-            request_type='POST',
-            request_headers=request_headers,
-            query_params=extra_request_data.get('query_params', {}),
+        if self.api_version == 'v0':
+            print('Report generation requested to: {}'.format(initial_report_request_url))
+
+            return request_handler(
+                request_url=initial_report_request_url,
+                request_data=report_generation_request_data,
+                request_type='POST',
+                request_headers=self.get_request_headers(),
+                query_params=extra_request_data.get('query_params', {}),
+            )
+
+        course_groups = get_course_groups(
```

Comment:
from here is it for v1?

Reply:
Yes, it involves everything related to the API v1.

Reply:
Is it possible that this script will read the API other future versions?, if so, maybe it could be refactored, otherwise I think the code is ok

Reply:
Yes, but how should it be refactored? What type of refactor?

Reply:
Separate in different methods for each implementation by API version, since in case of adding support for more versions in the method that is currently can grow more

Reply:
I agree, addressed. @luismorenolopera